emu_mps/mps_backend_impl.py

```diff
@@ -128,7 +141,12 @@ def __init__(self, mps_config: MPSConfig, pulser_data: PulserData):
         self.swipe_direction = SwipeDirection.LEFT_TO_RIGHT
         self.tdvp_index = 0
         self.timestep_index = 0
-        self.results = Results(atom_order=(), total_duration=self.target_times[-1])
+        self.results = Results(
+            atom_order=tuple(
+                optimat.permute_list(list(pulser_data.qubit_ids), self.qubit_permutation)
```

Comment:
I'm not sure if it's a good idea to store the permutation too. This field is part of the general API definition, and it's supposed to be a tuple of strings, i.e. just the `qubit_id`s in the permuted order.

Reply:
mypy changes a bit readability. The code essentially is

```
        old_order = list(pulser_data.qubit_ids)  # it's stored as a tuple
        permuted_order = optimat.permute_list(old_order, self.qubit_permutation)
        self.results = Results(
            atom_order=tuple(permuted_order),
            total_duration=self.target_times[-1],
        )
```

I store in `Results` only the permuted `qubit_ids`.

Did you mean smth else ?

Reply:
yes, you store in atom_order an object of type `tuple[list, list]` while it should be a tuple of type `tuple[str,...]` specifically, it should contain the first item from what you assign, cast to a tuple.

Reply:
have a look